blog/content/a-reintroduction-to-mkosi.md

```diff
@@ -0,0 +1,324 @@
+to switch to `root`.
+
+Note that as of the writing of this blog post this only applies to the
+`build` and `qemu` verbs. Booting the image in a `systemd-nspawn`
+container with `mkosi boot` still needs root privileges for now. We're
+hoping to fix this in an future systemd release.
+
+Regardless of whether you're running `mkosi` with root or without root,
+almost every tool we execute is invoked in a sandbox to isolate as much
+of the build process from the host as possible. For example, `/etc` and
+`/var` from the host are not available in this sandbox, to avoid host
+specific configuration affecting the build somehow.
+
+Because `systemd-repart` can build disk images without loop devices,
+`mkosi` can run from almost any environment, including containers. All
+that's needed is a UID range with 65536 UIDs available, either via
+running as the root user or via `/etc/subuid` and `newuidmap`. In a
+future systemd release, we're hoping to provide an alternative to
+`newuidmap` and `/etc/subuid` to allow running `mkosi` from all
+containers, even those with only a single UID available.
+
+## Supporting older distributions
+
+mkosi depends on very recent versions of various systemd tools (v254 or
+newer). To support older distributions, we implemented so called tools
+trees. In short, `mkosi` can first build a tools image for you that
+contains all required tools to build the actual image. This can be
+enabled by adding `ToolsTree=default` to your mkosi configuration.
+Building a tools image does not require a recent version of systemd.
+
+In the systemd mkosi configuration, we automatically use a tools tree if
+we detect your distribution does not have the minimum required systemd
+version installed.
+
+## Various other interesting features
+
+- Multiple images can be built at once by placing their configuration in
+  `mkosi.images/`. Dependencies between these images can be configured
+  with the `Dependencies=` setting. For example, you can first build a
+  custom initrd and then use it to build a bootable disk image
+- System extensions can be built using the `BaseTrees=` and `Overlay=`
+  settings. Point `BaseTrees=` to the base image, enable `Overlay=` and
+  set `Format=sysext` to have `mkosi` build an overlay on top of the
+  given base image and package it up as a system extension.
+- To sign any generated UKIs for secure boot, put your secure boot key
+  and certificate in `mkosi.key` and `mkosi.crt` and enable the
+  `SecureBoot=` setting. You can also run `mkosi genkey` to have `mkosi`
+  generate a key and certificate itself.
+- Profiles can be defined in the `mkosi.profiles/` directory. The
+  profile to use can be selected using the `Profile=` setting (or
+  `--profile=`) on the command line. Profiles allow selecting different
+  variants of the same image while still sharing most of the
+  configuration.
+- The `Ephemeral=` setting can be enabled to boot the image in an
```

Comment:
And this.

Reply:
Does this really needs its own paragraph? `Ephemeral=` is pretty simple

Reply:
I think it does, i.e. it's a separate topic that should be a separate para. Sometimes paragraphs may be very short. But it's not that important, feel free to ignore my comments.